### DISCUSSION TITLE
How to improve motion correction?

### DISCUSSION THREAD

**Discussion by jiwonkim2020** [miniscope/minian#194] [Q&A] [answered]
Hi. I'm working on a dataset of which neurons are sparse and somewhat weak (in terms of fluorescent intensity). Sometimes the output of motion correction seems not any better--I've attached an example image. I was wondering how I can improve this step. Is this related to the ksize of denoise or wnd of background removal (I've used 7 and 15, respectively)?

![image](https://user-images.githubusercontent.com/62463523/159807522-be593fcd-2812-4f5a-925f-f1661816f49b.png)

**Comment by phildong** [comment, marked as answer]
Hey so if you believe you're missing some cell after the preprocessing, then playing with ksize/wnd_size might help, otherwise they don't really play a strong role in motion correction. The real problem, as you noticed, is the sparsity of signals/landmarks for motion correction. There are usually different ways we deal with this depending on your data:
1. if you have some stable landmark, you can draw an ROI to use that landmark only for motion correction.
2. if you have some landmark that's removed by `remove_background` step, then you can do motion correction before the preprocessing steps. This is oftentimes used in conjunction with 1
3. if all fails but you know there's really not much motion in your data to begin with, you can skip motion correction altogether.

**Reply by jiwonkim2020** [reply]
Thank you, Phil! This is helpful. I'll try your suggestions.
